core/engine/src/object/operations.rs

```diff
@@ -931,12 +936,25 @@ impl JsObject {
             method,
             PrivateElement::Method(_) | PrivateElement::Accessor { .. }
         ));
+
         let (getter, setter) = if let PrivateElement::Accessor { getter, setter } = method {
             (getter.is_some(), setter.is_some())
         } else {
             (false, false)
         };
 
+        if !self.extensible() {
```

Comment:
Suggestion: Same here:

```suggestion
        // 2. If O.[[Extensible]] is false, throw a TypeError exception.
        //
        // NOTE: From <https://tc39.es/proposal-nonextensible-applies-to-private/#sec-privatemethodoraccessoradd>
        if !self.extensible() {
```